libs/MeshKernel/src/Operations.cpp

```diff
@@ -1232,6 +1232,80 @@ namespace meshkernel
         return circumcenter;
     }
 
+    UInt CountNumberOfValidEdges(const std::vector<UInt>& edgesNumFaces, const UInt numNodes)
```

Comment:
UInt CountNumberOfValidEdges(const std::vector<UInt>& edgesNumFaces, UInt numNodes)
{
    numNodes = std::min(numNodes, static_cast<UInt>(edgesNumFaces.size())); // Avoid out-of-bounds access
    return std::count(edgesNumFaces.begin(), edgesNumFaces.begin() + numNodes, 2);
}